component/prometheus/kubernetes/podmonitors/podMonitor.go

```diff
@@ -0,0 +1,269 @@
+package podmonitors
+
+// SEE https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	namespacelabeler "github.com/prometheus-operator/prometheus-operator/pkg/namespace-labeler"
+	commonConfig "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	promk8s "github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/model/relabel"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	regexFilterRunning = relabel.MustNewRegexp("(Failed|Succeeded)")
+	regexTrue          = relabel.MustNewRegexp("true")
+	regexAnything      = relabel.MustNewRegexp("(.+)")
+)
+
+func (cg *configGenerator) generatePodMonitorConfig(m *v1.PodMonitor, ep v1.PodMetricsEndpoint, i int) (cfg *config.ScrapeConfig, err error) {
```

Comment:
nit: I always find it surprising and hard to navigate the code base when methods for a struct are split across multiple files, unless the file names indicate that it's a continuation of something. 

I would generally expect this file to be called `config_generator_podmonitor.go`. I also wouldn't use camelCase for any of file names, since no other file in this code base is named that way, and I like being consistent across the board.